### DISCUSSION TITLE
Table Column hidden method cannot get the current record throws error

### DISCUSSION THREAD

**Discussion by khanakia** [filamentphp/filament#14201] [Help]
### Package

Table builder

### Package Version

v3.2.69

### How can we help you?
I am following this docs https://filamentphp.com/docs/3.x/tables/columns/advanced#injecting-multiple-utilities
to hide the Column if posts status is unpublished

```php
 Tables\Columns\TextColumn::make('view')
      ->default('View')
      ->hidden(function (Post $record) {
          return $record->status == 'unpublished';
      })
```

Getting error 
```
App\Filament\Resources\PostResource::App\Filament\Resources\{closure}(): Argument #1 ($record) must be of type App\Models\Post, null given, called in /Volumes/D/www/projects/aa/test/vendor/filament/support/src/Concerns/EvaluatesClosures.php on line 35
```

**Comment by leandrocfe** [comment]
You can't show/hide a column based on its rows. You can use `->disabled()` instead or `->formatStateUsing()`

https://filamentphp.com/docs/3.x/tables/columns/text/#rendering-html

**Reply by khanakia** [reply]
disabled not working it never runs the disabled method at all

I just do not want the VIEW link to show for the Post if it's unpublished. 


```php
 Tables\Columns\TextColumn::make('view')
                    ->default('View')
                    ->disabled(function (Post $record) {
                        dd($record);
                        return $record->status == 'unpublished';
                    })
```

![Screenshot 2024-09-11 at 2 24 00 PM](https://github.com/user-attachments/assets/e4504a00-1a59-4be1-bdc9-d4c298fc0c4c)

**Reply by leandrocfe** [reply]
```php
TextColumn::make('view')
    ->getStateUsing(fn (Post $record): string => $record->status === 'published' ? 'View' : '')
```

**Reply by khanakia** [reply]
It removed the label but I can still click the empty area and it redirects


https://github.com/user-attachments/assets/d3b5b36c-d041-4493-900e-d5088de835ec


![Screenshot 2024-09-12 at 5 49 37 PM](https://github.com/user-attachments/assets/b987d69b-374b-4895-a20b-db8745fff0a0)

**Reply by leandrocfe** [reply]
Could you share the code?

**Reply by khanakia** [reply]
```php
Tables\Columns\TextColumn::make('view')
                    // ->default('View')
                    ->getStateUsing(fn(Post $record): string => $record->status === 'published' ? 'View' : '')
                    // ->disabled(function (Post $record) {
                    //     dd($record);
                    //     return $record->status == 'unpublished';
                    // })
                    ->url(function ($record) {
                        return $record->url;
                    }, true)->toggleable(),
```

**Reply by leandrocfe** [reply]
```php
->getStateUsing(fn (Post $record): ?string => $record->status === 'published' ? 'View' : null)
->url(fn (Post $record): ?string => $record->status ==='published' ? $record->url : null)
```

**Comment by griggsk** [comment]
The error `Argument #1 ($record) must be of type App\Models\Post, null given`  is telling you that you do not have a Post record set. Did you set it in mount() or in ->relationship() or ...?

Once you have it set you can do:
->hidden(fn (): bool => $this->post->status === 'published' ? 'View' : '')

**Reply by khanakia** [reply]
Why I have to set in the mount I am using Filament Resource? All I want is not to show the View link in the table if specific post is unpublished

you can check the attached file

[PostResource.php.txt](https://github.com/user-attachments/files/17001980/PostResource.php.txt)
